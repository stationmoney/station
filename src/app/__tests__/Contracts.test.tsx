import { render, screen, waitFor } from "@testing-library/react"
import React, { useState as useStateMock } from "react"
import Contract from "pages/contract/Contract"
import { mount, shallow } from "enzyme"
import { useQuery } from "react-query"
import userEvent from "@testing-library/user-event"
import { MemoryRouter } from "react-router-dom"
import { mockContract } from "./__mocks__/Contract.mock"
import { mockNetworks } from "./__mocks__/Networks.mock"

function renderComponent() {
  return render(
    <MemoryRouter>
      <Contract />
    </MemoryRouter>
  )
}

jest.mock("../../data/wallet", () => {
  const mockUseAddress = () => {
    return "terra111111111111111111111111111111111111111"
  }

  const mockUseNetwork = () => {
    return mockNetworks
  }

  const mockUseNetworkName = () => {
    return "testnet"
  }

  return {
    useAddress: mockUseAddress,
    useNetwork: mockUseNetwork,
    useNetworkName: mockUseNetworkName,
  }
})

jest.mock("react-query", () => ({
  useQuery: jest.fn().mockReturnValue({
    data: Object,
    isLoading: Boolean,
    error: Object,
  }),
}))

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "/contract",
    search: "",
    state: null,
    hash: "",
    key: "pbbp8kkj",
  }),
}))

jest.mock("../../data/queries/wasm", () => {
  const mockUseContractInfo = (address: any) => {
    return mockContract
  }

  return {
    useContractInfo: mockUseContractInfo,
  }
})

// describe('Contract component matches snapshots', () => {
//   it('matches original component', () => {
//     useQuery.mockReturnValue({
//       data: mockContract,
//       isLoading: false,
//       error: {},
//     });
//     const { asFragment } = renderComponent();
//     expect(asFragment()).toMatchSnapshot();
//   });

//   it('does not find contract', async () => {
//     useQuery.mockReturnValue({
//       data: mockContract,
//       isLoading: false,
//       error: {},
//     });
//     const { asFragment } = renderComponent();

//     // Search for contract.
//     const contractSearch = screen.getByRole('textbox');
//     await userEvent.type(contractSearch, 'terra');

//     // Ensure snapshot matches user augmented component.
//     expect(asFragment()).toMatchSnapshot();
//   });
// });

describe("Find contract with address", () => {
  it("finds contract", async () => {
    useQuery.mockReturnValue({
      result: mockContract,
      isLoading: false,
      error: false,
    })
    const { asFragment } = renderComponent()

    // Search for contract.
    const contractSearch = screen.getByRole("textbox")
    await userEvent.type(
      contractSearch,
      "terra18qr46nvqnymd4wl4md58kgdz699xcwcf6h7dd0"
    )

    // Ensure snapshot matches user augmented component.
    expect(asFragment()).toMatchSnapshot()
  })
})
