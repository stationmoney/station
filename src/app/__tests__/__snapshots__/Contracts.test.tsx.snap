// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Find contract with address finds contract 1`] = `
<DocumentFragment>
  <article
    class="page"
  >
    <div
      class="container grid"
    >
      <header
        class="header"
      >
        <div
          class="titleWrapper"
        >
          <h1
            class="title"
          >
            Contract
          </h1>
        </div>
        <div
          class="actions end"
        >
          <a
            class="button small primary"
            href="/contract/store"
          >
            Upload
          </a>
          <a
            class="button small primary"
            href="/contract/instantiate"
          >
            Instantiate
          </a>
        </div>
      </header>
      <section
        class="main"
      >
        <div
          class="grid"
          style="gap: 20px;"
        >
          <main
            class="contract__main"
          >
            <section
              class="contract__search"
            >
              <div
                class="wrapper search"
                style="margin: 0px;"
              >
                <input
                  autocomplete="off"
                  class="input"
                  inputmode="search"
                  value="terra"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
                  data-testid="SearchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </div>
            </section>
            <section
              class="contract__body"
            >
              <article
                class="card default"
              >
                <section
                  class="main"
                >
                  <div
                    class="grid component"
                    style="gap: 8px;"
                  >
                    <div
                      class="flex icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                        data-testid="ManageSearchIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 9H2V7h5v2zm0 3H2v2h5v-2zm13.59 7-3.83-3.83c-.8.52-1.74.83-2.76.83-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5c0 1.02-.31 1.96-.83 2.75L22 17.59 20.59 19zM17 11c0-1.65-1.35-3-3-3s-3 1.35-3 3 1.35 3 3 3 3-1.35 3-3zM2 19h10v-2H2v2z"
                        />
                      </svg>
                    </div>
                    <section
                      class="content"
                    >
                      Search by contract address
                    </section>
                  </div>
                </section>
              </article>
            </section>
          </main>
        </div>
      </section>
    </div>
  </article>
</DocumentFragment>
`;
